### ISSUE TITLE
Evaluate Keywords fails with list comprehension

### ISSUE THREAD

**Issue by RoscoeReeseHiggings** [robotframework/robotframework#3173]
In this testcase, a nested list comprehension is evaluated differently than when using the python eval function:

```robot
*** Variables ***

@{a}=  a1  a2
@{b}=   b1  b2

&{data}=  a=${a}  b=${b}     

*** Test Cases ***

test
    Log  ${data}
    Evaluate  [(x, y) for x in ${data} for y in ${data}[x]]
```
fails with the message log:
 
INFO : {'a': ['a1', 'a2'], 'b': ['b1', 'b2']}
FAIL : Dictionary '${data}' has no key 'x'.

Trying to do the same in pyhton

```python
>>> data = {'a': ['a1', 'a2'], 'b': ['b1', 'b2']}
>>> [(x, y) for x in data for y in data[x]]
[('a', 'a1'), ('a', 'a2'), ('b', 'b1'), ('b', 'b2')]
```

or using the eval function:

```python
>>> eval("[(x, y) for x in {'a': ['a1', 'a2'], 'b': ['b1', 'b2']} for y in {'a': ['a1', 'a2'], 'b': ['b1', 'b2']}[x]]")
[('a', 'a1'), ('a', 'a2'), ('b', 'b1'), ('b', 'b2')]
```

produces the desired output.
Reading the docs I could make it work by using the keyword argument namespace:

```robot
test    
    ${ns}=  Create Dictionary  data=${data}  
    ${x}=  Evaluate  [(x, y) for x in data for y in data[x]]  namespace=${ns}
    Log  ${x}
```
logs INFO :  [('a', 'a1'), ('a', 'a2'), ('b', 'b1'), ('b', 'b2')]

I have problems understanding why the first robot example doesn't work,  and if this is the correct behaviour.

**Comment by pekkaklarck** [comment]
Did you read the section about evaluating expressions from the beginning of the library docs?

**Comment by RoscoeReeseHiggings** [comment]
Yes, I did. But after looking at it and my code a second time, I realize where I went wrong:
In the expression 
```robot
Evaluate  [(x, y) for x in ${data} for y in ${data}[x]]
```
``` ${data}[x]```  is replaced with the string representation of its variable value before evaluation in python, I wrongly assumed ```${data}``` would be evaluated first, and the indexing would happen in python.
What I intended can be achieved by escaping the opening square bracket
```robot
Evaluate  [(x, y) for x in ${data} for y in ${data}\[x]]
```

Thanks for the suggestion!
